File: 프로그래머스/lv3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    for i in range(len(triangle) - 1, 0, -1):
        for j in range(1, i + 1):
            if triangle[i][j - 1] <= triangle[i][j]:
                triangle[i - 1][j - 1] += triangle[i][j]
            else:
                triangle[i - 1][j - 1] += triangle[i][j - 1]
    return triangle[0][0]

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
class Solution {
    public int solution(int[][] triangle) {
        for (int x = triangle.length - 2; x >= 0; x--) {
            for (int y = 0; y < triangle[x].length; y++) {
                triangle[x][y] += Math.max(triangle[x + 1][y], triangle[x + 1][y + 1]);
            }
        }
        
        return triangle[0][0];
    }
}